src/main/java/com/auth0/client/auth/AuthAPI.java

```diff
@@ -54,6 +56,7 @@ public AuthAPI(String domain, String clientId, String clientSecret) {
         if (baseUrl == null) {
             throw new IllegalArgumentException("The domain had an invalid format and couldn't be parsed as an URL.");
         }
+        this.httpUrl = HttpUrl.parse(baseUrl);
```

Comment:
`createBaseUrl` already uses `HttpUrl.parse`, so maybe make that return that httpUrl directly?
then `baseUrl` (if really required) can be obtained from `baseUrl = httpUrl.toString()`